File: Dddml.Wms.JavaCommon/src/org/dddml/wms/domain/YearPlanMvoStateEventId.java
package org.dddml.wms.domain;

import java.io.Serializable;

public class YearPlanMvoStateEventId implements Serializable
{
    private YearPlanId yearPlanId = new YearPlanId();

    public YearPlanId getYearPlanId()
    {
        return this.yearPlanId;
    }

    public void setYearPlanId(YearPlanId yearPlanId)
    {
        this.yearPlanId = yearPlanId;
    }

    private Long personVersion;

    public Long getPersonVersion()
    {
        return this.personVersion;
    }

    public void setPersonVersion(Long personVersion)
    {
        this.personVersion = personVersion;
    }

    public String getYearPlanIdPersonalNameFirstName()
    {
        return getYearPlanId().getPersonalName().getFirstName();
    }

    public void setYearPlanIdPersonalNameFirstName(String yearPlanIdPersonalNameFirstName)
    {
        getYearPlanId().getPersonalName().setFirstName(yearPlanIdPersonalNameFirstName);
    }

    public String getYearPlanIdPersonalNameLastName()
    {
        return getYearPlanId().getPersonalName().getLastName();
    }

    public void setYearPlanIdPersonalNameLastName(String yearPlanIdPersonalNameLastName)
    {
        getYearPlanId().getPersonalName().setLastName(yearPlanIdPersonalNameLastName);
    }

    public Integer getYearPlanIdYear()
    {
        return getYearPlanId().getYear();
    }

    public void setYearPlanIdYear(Integer yearPlanIdYear)
    {
        getYearPlanId().setYear(yearPlanIdYear);
    }

    public YearPlanMvoStateEventId()
    {
    }

    public YearPlanMvoStateEventId(YearPlanId yearPlanId, Long personVersion)
    {
        this.yearPlanId = yearPlanId;
        this.personVersion = personVersion;
    }

    @Override
    public boolean equals(Object obj)
    {
        if (obj == this) {
            return true;
        }
        if (obj == null || obj.getClass() != this.getClass()) {
            return false;
        }

        YearPlanMvoStateEventId other = (YearPlanMvoStateEventId)obj;
        return true 
            && (yearPlanId == other.yearPlanId || (yearPlanId != null && yearPlanId.equals(other.yearPlanId)))
            && (personVersion == other.personVersion || (personVersion != null && personVersion.equals(other.personVersion)))
            ;
    }

    @Override
    public int hashCode()
    {
        int hash = 0;
        if (this.yearPlanId != null) {
            hash += 13 * this.yearPlanId.hashCode();
        }
        if (this.personVersion != null) {
            hash += 13 * this.personVersion.hashCode();
        }
        return hash;
    }

}

File: Dddml.Wms.HttpServices.PhpClient/Dddml/Wms/Domain/YearPlanMvoStateEventId.php
<?php

namespace Dddml\Wms\Domain;

use JMS\Serializer\Annotation\Type;
use Dddml\Serializer\Type\Long;
use Dddml\StringIdInterface;

class YearPlanMvoStateEventId implements StringIdInterface
{
    /**
     * @Type("Dddml\Wms\Domain\YearPlanId")
     */
    private $yearPlanId;

    /**
     * @return YearPlanId
     */
    public function getYearPlanId()
    {
        if(!$this->yearPlanId) {
            $this->yearPlanId = new YearPlanId(); 
        }
        return $this->yearPlanId;
    }

    /**
     * @param YearPlanId $yearPlanId
     */
    public function setYearPlanId($yearPlanId)
    {
        $this->yearPlanId = $yearPlanId;
    }

    /**
     * @Type("Dddml\Serializer\Type\Long")
     */
    private $personVersion;

    /**
     * @return Long
     */
    public function getPersonVersion()
    {
        return $this->personVersion;
    }

    /**
     * @param Long $personVersion
     */
    public function setPersonVersion($personVersion)
    {
        $this->personVersion = $personVersion;
    }



    /**
     * @var YearPlanMvoStateEventIdFlattenedDto
     */
    private $idFlattenedDto;

    /**
     * @return string
     */
    public function toString()
    {
        if (!$this->idFlattenedDto) {
            $this->idFlattenedDto = new YearPlanMvoStateEventIdFlattenedDto($this);
        }

        return $this->idFlattenedDto->toString();
    }

    /**
     * @param string $idStr
     *
     * @return YearPlanMvoStateEventId
     */
    public static function createFromString($idStr)
    {
        return (new YearPlanMvoStateEventIdFlattenedDto())
            ->fromString($idStr)
            ->toYearPlanMvoStateEventId();
    }


}

